ddclient

```diff
@@ -2307,6 +2306,25 @@ sub get_ip {
     return $ip;
 }
 
+######################################################################
+## is_ipv4() validates if string is valid IPv4 address and only
+## a valid IPv4 address (no preceding or trailing spaces/characters)
+######################################################################
+sub is_ipv4 {
+    my ($value) = @_;
+    return (length($value) != 0) && ($value eq ipv4_match($value));
```

Comment:
If `$value` is `undef`, or if there's no IPv4 address in `$value`, then this will log a warning. How about this instead:

```suggestion
    return (($value // '') ne '') && ($value eq (ipv4_match($value) // ''));
```

Reply:
I read that testing for length equal zero is more efficient than a string compare to test for an empty string.  Also, should not be necessary to test for undefined after the && as it will never get there.  So I suggest the following.
```
return (length($value // '') != 0) && ($value eq ipv4_match($value));
```

And, btw, thanks for pointing this out...  I have turned on warnings in my perl test script so that I catch things like this.

Reply:
> I read that testing for length equal zero is more efficient than a string compare to test for an empty string.

That was true in older versions of Perl, but I don't know if it's still true—I couldn't find any recent discussion about it. Even if `eq ''` is slower than a length check, using a length check feels like a premature micro-optimization.

In general I prefer to err on the side of human readability (and thus maintainability) rather than performance, unless performance is critical. In this case I'm OK with it either way; a length check is idiomatic enough that it won't surprise anyone.